### PR TITLE
Add initial mysql compatibility document

### LINKED ISSUE #222: Document "MySQL Compatibility"
Add a page under "Reference" which clearly explains MySQL compatibility story.  Rename https://vitess.io/docs/reference/mysql-server-protocol/ to fit in this page.

### PR DESCRIPTION
**Pull request by morgo**
Fixes #222

This is an initial version, I'm sure we can do better in future.

Signed-off-by: Morgan Tocker <tocker@gmail.com>

### CODE REVIEW

File: content/en/docs/reference/mysql-compatibility.md [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+Vitess provides `READ-COMMITTED` semantics when executing cross-shard queries. This differs to MySQL, which defaults to `REPEATABLE-READ`. 
+
+## SQL Syntax
+
+The following describes some of the major differences in SQL Syntax handling between Vitess and MySQL. For a list of unsupported queries, check out the [test-suite cases](https://github.com/vitessio/vitess/blob/master/go/vt/vtgate/planbuilder/testdata/unsupported_cases.txt).
+
+### Join Queries
+
+Vitess supports `INNER JOIN` including cross-shard joins. `LEFT JOIN` and natural join are not yet supported in sharded keyspaces.
```
Comment (sougou) [inline content/en/docs/reference/mysql-compatibility.md]: left join is supported.

File: content/en/docs/reference/mysql-compatibility.md [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+### Aggregation
+
+Vitess supports a subset of `GROUP BY` operations, including cross-shard operations. The VTGate servers are capable of scatter-gather operations, but can only stream results. Thus, a query that performs a `GROUP BY colx ORDER BY coly` will refuse to execute as this requires the intermediate results to be materialized.
```
Comment (sougou) [inline content/en/docs/reference/mysql-compatibility.md]: `group by colx order by coly ` is now supported. The requirement is that the in-memory sub-result does not exceed the vtgate limit.

File: content/en/docs/reference/mysql-compatibility.md [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+### Subqueries
+
+Vitess supports a subset of subqueries. For example, a subquery combined with a `GROUP BY` operation is not supported.
+
+### Window Functions and CTEs
+
+Vitess does not yet support Window Functions or Common Table Expressions.
+
+### Cross-shard Update
```
Comment (sougou) [inline content/en/docs/reference/mysql-compatibility.md]: Cross-shard Transactions

File: content/en/docs/reference/mysql-compatibility.md [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+
+By default, Vitess does not support atomic modifications across shards. While Vitess can support this with the use of [Two-Phase Commit](../two-phase-commit), it is usually recommended to design the VSchema in such a way that cross-shard modifications are not required.
```
Comment (sougou) [inline content/en/docs/reference/mysql-compatibility.md]: transactions that span across shards.

File: content/en/docs/reference/mysql-compatibility.md [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+* `CREATE TEMPORARY TABLE`
+* `SET @var=x`
+
+The exception to this, is that Vitess maintains a whitelist statements that MySQL connectors may use when they first connect to Vitess. Vitess will ignore these noisy statements when it knows it is safe to do so.
+
+## Character Set and Collation
+
+Vitess only supports `utf8` and variants such as `utf8mb4`.
+
+## SQL Mode
```
Comment (sougou) [inline content/en/docs/reference/mysql-compatibility.md:62]: This is not a restriction any more. It was only required for SBR.

Comment (morgo) [reply content/en/docs/reference/mysql-compatibility.md:62]: I have tried to change the sql-mode to MySQL's default, and the test-suite fails. There are some modes like `NO_ZERO_DATE`, `NO_ZERO_IN_DATE` which at least the suite doesn't support yet. I'd like to leave this in here until we fix it if that's okay with you.

File: content/en/docs/reference/mysql-compatibility.md (hunk 1 above)
Comment (morgo) [reply content/en/docs/reference/mysql-compatibility.md]: Thanks, I've clarified the case a little bit better.

**Review by sougou — APPROVED** [review]

**Pull request merged**